pkg/controller/namespace/deletion/namespaced_resources_deleter.go

```diff
@@ -480,35 +480,47 @@ func (d *namespacedResourcesDeleter) deleteAllContentForGroupVersionResource(
 // deleteAllContent will use the dynamic client to delete each resource identified in groupVersionResources.
 // It returns an estimate of the time remaining before the remaining resources are deleted.
 // If estimate > 0, not all resources are guaranteed to be gone.
-func (d *namespacedResourcesDeleter) deleteAllContent(namespace string, namespaceDeletedAt metav1.Time) (int64, error) {
+func (d *namespacedResourcesDeleter) deleteAllContent(ns *v1.Namespace) (int64, error) {
+	namespace := ns.Name
+	namespaceDeletedAt := *ns.DeletionTimestamp
 	var errs []error
+	conditionUpdater := namespaceConditionUpdater{}
 	estimate := int64(0)
 	klog.V(4).Infof("namespace controller - deleteAllContent - namespace: %s", namespace)
 
 	resources, err := d.discoverResourcesFn()
 	if err != nil {
 		// discovery errors are not fatal.  We often have some set of resources we can operate against even if we don't have a complete list
 		errs = append(errs, err)
+		conditionUpdater.ProcessDiscoverResourcesErr(err)
 	}
 	// TODO(sttts): get rid of opCache and pass the verbs (especially "deletecollection") down into the deleter
 	deletableResources := discovery.FilteredBy(discovery.SupportsAllVerbs{Verbs: []string{"delete"}}, resources)
 	groupVersionResources, err := discovery.GroupVersionResources(deletableResources)
 	if err != nil {
 		// discovery errors are not fatal.  We often have some set of resources we can operate against even if we don't have a complete list
 		errs = append(errs, err)
+		conditionUpdater.ProcessGroupVersionErr(err)
 	}
 	for gvr := range groupVersionResources {
 		gvrEstimate, err := d.deleteAllContentForGroupVersionResource(gvr, namespace, namespaceDeletedAt)
 		if err != nil {
 			// If there is an error, hold on to it but proceed with all the remaining
 			// groupVersionResources.
 			errs = append(errs, err)
+			conditionUpdater.ProcessDeleteContentErr(err)
 		}
 		if gvrEstimate > estimate {
 			estimate = gvrEstimate
 		}
 	}
+
 	if len(errs) > 0 {
+		if hasChanged := conditionUpdater.Update(ns); hasChanged {
+			if _, err = d.nsClient.UpdateStatus(ns); err != nil {
+				utilruntime.HandleError(fmt.Errorf("Couldn't update status condition for namespace %q: %v", namespace, err))
```

Comment:
Let's use `errors.Wrapf` here.

Reply:
Given this package uses `fmt.Errorf()` multiple times and does not use `errors.Wrapf()` or `errors.Wrap()`, it may be better to stay in sync with the package conventions, what do you think?

Also `pkg/controller/` seem to be more old-school :)
```
$ grep pkg/controller/ -nrie Wrapf --include=*.go | wc -l
0
$ grep pkg/controller/ -nrie Errorf --include=*.go | wc -l
1771
```

Reply:
Makes sense.